runtime/runtime_test.go

```diff
@@ -7706,3 +7706,111 @@ func TestRuntimeTypeMismatchErrorMessage(t *testing.T) {
 	require.ErrorContains(t, err, "expected type `A.0000000000000002.Foo.Bar`, got `A.0000000000000001.Foo.Bar`")
 
 }
+
+func TestRuntimeErrorExcerpts(t *testing.T) {
+
+	t.Parallel()
+
+	rt := newTestInterpreterRuntime()
+
+	script := []byte(`
+    pub fun main(): Int {
+        // fill lines so the error occurs on lines 9 and 10
+        // 
+        // 
+        //
+        //
+        let a = [1,2,3,4]
+        return a
+            .firstIndex(of: 5)!
+    }
+    `)
+
+	runtimeInterface := &testRuntimeInterface{
+		getAccountBalance:          noopRuntimeUInt64Getter,
+		getAccountAvailableBalance: noopRuntimeUInt64Getter,
+		getStorageUsed:             noopRuntimeUInt64Getter,
+		getStorageCapacity:         noopRuntimeUInt64Getter,
+		accountKeysCount:           noopRuntimeUInt64Getter,
+		storage:                    newTestLedger(nil, nil),
+	}
+
+	nextTransactionLocation := newTransactionLocationGenerator()
+
+	_, err := rt.ExecuteScript(
+		Script{
+			Source: script,
+		},
+		Context{
+			Interface: runtimeInterface,
+			Location:  nextTransactionLocation(),
+		},
+	)
+	require.Error(t, err)
+
+	errorString := `Execution failed:
+error: unexpectedly found nil while forcing an Optional value
+  --> 0000000000000000000000000000000000000000000000000000000000000000:9:15
+   |
+ 9 |         return a
+10 |             .firstIndex(of: 5)!
+   |                ^^^^^^^^^^^^^^^^
```

Comment:
From what I can gather, the indicator still starts at the start pos column (`a`). We can maybe improve that in a subsequent PR, e.g. by rendering indicators for each line